scrapy/spiderloader.py

```diff
@@ -24,15 +24,17 @@ def __init__(self, settings):
         self._load_all_spiders()
 
     def _check_name_duplicates(self):
-        dupes = ["\n".join("  {cls} named {name!r} (in {module})".format(
-                                module=mod, cls=cls, name=name)
-                           for (mod, cls) in locations)
-                 for name, locations in self._found.items()
-                 if len(locations) > 1]
+        dupes = []
+        for name, locations in self._found.items():
+            dupes.extend([
+                "  {cls} named {name!r} (in {module})".format(module=mod, cls=cls, name=name)
+                for mod, cls in locations
+            ])
+
```

Comment:
This might be a little bit more verbose, but way more readable IMHO. Personally, I find nested `for` lops hard to follow.

Reply:
It seems `if len(locations) > 1` check is removed in the new code. How does it work? Do we have tests for it?

Reply:
Thanks for noticing this. I thought the list would be empty in that case, but it's `>1` and not `>=1`. I'll fix it.

Reply:
I was thinking that's the purpose of the function - warn in case of duplicates, i.e. a single name with 2+ locations; that's bad that our tests don't catch it.

Reply:
Makes sense. I'll open a issue to keep track of this, I don't think we should add it in this PR.

Reply:
Done: #4549. I skipped the issue template entirely :sweat_smile: